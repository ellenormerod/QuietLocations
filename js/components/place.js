import React, { Component } from 'react';
import {
  Text,
  View,
  FlatList
} from 'react-native';
import { ListItem, List, Card, Rating } from 'react-native-elements'
import { connect } from 'react-redux'
import { fetchReviews } from '../store'
import { Actions } from 'react-native-router-flux';


export class Place extends Component {
  componentDidMount() {
    this.props.loadReviews(this.props.item.id)
  }

  renderSeparator = () => {
    return (
      <View
        style={{
          height: 1,
          width: '86%',
          backgroundColor: '#CED0CE'
        }}
      />
    )
  }

  render() {
    const { item, reviews } = this.props
    return (
       <View>
       <Card title={item.name}>
          <Text h3>
            {item.location}
          </Text>
          <Rating
            showRating
            type="star"
            fractions={1}
            startingValue={+item.noise}
            readonly
            imageSize={30}
            // onFinishRating={review.noise}
            style={{ paddingVertical: 10 }}
          />
        </Card>
        <List containerStyle={{ borderTopWidth: 0, borderBottomWidth: 0 }}>
         <FlatList
        data={reviews}
        renderItem={({ item }) => (
          <ListItem
                title={item.title}
                subtitle={item.comments}
                containerStyle={{ borderBottomWidth: 0 }}
                onPress={() => Actions.review({ item })}
              />
            )}
            keyExtractor={item => item.title}
            ItemSeparatorComponent={this.renderSeparator}
          />
        </List>
      </View>
    )
  }
}

const mapState = state => {
  return {
    reviews: state.reviews
  }
}

const mapDispatch = dispatch => {
  return {
    loadReviews(id) {
      dispatch(fetchReviews(id))
    }
  }
}

export default connect(mapState, mapDispatch)(Place)
