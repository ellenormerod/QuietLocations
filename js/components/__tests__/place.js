import React from 'react';
import { Place } from '../place';
import renderer from 'react-test-renderer';

const fakeProps = {
  item: {
    name: 'Great Bar',
    location: '123 W Cool Street St. Chicago, IL',
    noise: 3
  },
  loadReviews: () => { }
}


it('renders Places using Snapshot', () => {
  expect(renderer.create(
    <Place {...fakeProps} />
  )).toMatchSnapshot()
})
