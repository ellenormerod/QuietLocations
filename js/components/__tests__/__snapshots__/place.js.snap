// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Place using Snapshot 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#e1e8ee",
          "borderWidth": 1,
          "margin": 15,
          "marginBottom": 0,
          "padding": 15,
          "shadowColor": "rgba(0,0,0, .2)",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
          undefined,
        ]
      }
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={undefined}
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "#43484d",
                  "fontSize": 17.5,
                  "fontWeight": "bold",
                  "marginBottom": 15,
                  "textAlign": "center",
                },
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Great Bar
        </Text>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#e1e8ee",
                "height": 1,
              },
              Array [
                Object {
                  "marginBottom": 15,
                },
                undefined,
              ],
            ]
          }
        />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        h3={true}
      >
        123 W Cool Street St. Chicago, IL
      </Text>
      <View
        pointerEvents="none"
        style={
          Object {
            "paddingVertical": 10,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "column",
              "justifyContent": "center",
              "paddingBottom": 5,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingBottom": 5,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Object {
                    "color": "#34495e",
                    "fontFamily": "Trebuchet MS",
                    "fontSize": 15,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Rating: 
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "fontFamily": "Trebuchet MS",
                      "fontSize": 30,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#f1c40f",
                    },
                  ],
                ]
              }
            >
              3
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Object {
                    "color": "#34495e",
                    "fontFamily": "Trebuchet MS",
                    "fontSize": 18,
                    "textAlign": "center",
                  },
                ]
              }
            >
              /
              5
            </Text>
          </View>
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Object {
                    "alignItems": "center",
                    "color": "#34495a",
                    "fontFamily": "Trebuchet MS",
                    "fontSize": 12,
                    "justifyContent": "center",
                    "textAlign": "center",
                  },
                ]
              }
            >
              (readonly)
            </Text>
          </View>
        </View>
        <View
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "left": 0,
                "position": "absolute",
                "top": 0,
              }
            }
          >
            <View
              collapsable={undefined}
              style={
                Object {
                  "backgroundColor": "#f1c40f",
                  "height": 30,
                  "width": 90,
                }
              }
            />
            <View
              collapsable={undefined}
              style={
                Object {
                  "backgroundColor": "white",
                  "height": 30,
                  "width": 60,
                }
              }
            />
          </View>
          <View
            style={undefined}
          >
            <Image
              source={
                Object {
                  "testUri": "../../react-native-elements/src/rating/images/star.png",
                }
              }
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <View
            style={undefined}
          >
            <Image
              source={
                Object {
                  "testUri": "../../react-native-elements/src/rating/images/star.png",
                }
              }
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <View
            style={undefined}
          >
            <Image
              source={
                Object {
                  "testUri": "../../react-native-elements/src/rating/images/star.png",
                }
              }
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <View
            style={undefined}
          >
            <Image
              source={
                Object {
                  "testUri": "../../react-native-elements/src/rating/images/star.png",
                }
              }
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <View
            style={undefined}
          >
            <Image
              source={
                Object {
                  "testUri": "../../react-native-elements/src/rating/images/star.png",
                }
              }
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#bbb",
          "borderTopWidth": 1,
          "marginTop": 20,
        },
        Object {
          "borderBottomWidth": 0,
          "borderTopWidth": 0,
        },
      ]
    }
  >
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      data={undefined}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
